src/pages/services/aem-cf-editor/api/rte-toolbar/index.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Rich Text Editor Toolbar - AEM Content Fragments Editor Extensibility
+description: Learn what is possible to extend and customize in AEM Content Fragments Editor
+contributors:
+  - https://git.corp.adobe.com/dx-devex-acceleration/uix-docs
+---
+
+# Rich Text Editor Toolbar
+
+A Rich Text Editor is a core part of content editing software. With toolbar extensions you can add new buttons which modify the content:
+```js
+{
+  rte: {
+    getButtons: () => ([{
+      id: "test-id",
+      text: "Test button", //optional text near button
+      tooltip: "Button alt text",
+      icon: 'Airplane', //Spectrum workflow icon code from https://spectrum.adobe.com/page/icons/
+      onClick: () => {
+        return [{
+          type: "replaceContent",
+          value: state.html + "<p> some new HTML </p>"
+        }]
+      }
+    }])
+  }
+}
+```
+![](./rte-content-extension.gif)
+
+
+## API Reference
+
+**Extension API**
+
+| Field | Type | Required | Description |
+| ----- | ---- | -------- | ----------- |
+| id | `string` | ✔️    | **Must be unique** across all extensions. Consider adding a vendor prefix to this field. |
+| text | `string` |     | A button text |
+| tooltip | `string` | ✔️    | A text that will be rendered on mouse over event |
+| icon | `string` |     | An icon field accepts workflow icon code from @spectrum-icons library - https://spectrum.adobe.com/page/icons/ |
+| onClick | `callback(state): Instruction[]` |  ✔️  |  A callback for a button `onClick` event. The callback receives the editor `state` object as an argument and may return a list of instructions |
+
+
+**`onClick` `state` argument API**
+
+| Field | Type |  Description |
+| ----- | ---- | ----------- |
+| html | `string` | HTML representation of editor's content |
+| text | `string` | Text representation of editor's content |
+| selectedHtml | `string` | HTML representation of selected editor's content. Empty when there is no selection |
+| selectedText | `string` | Text representation of selected editor's content. Empty when there is no selection |
+
+
+**`onClick` instructions API**
+
+`onClick` method **MUST** return a list of the following instructions (could be an empty list):
+
+| Instruction | Value |  Description |
+| ----- | ---- | ----------- |
+| `replaceContent` | `string` | Replaces current editor content with a content provided in `value` property |
```

Comment:
Would it be hard to add `replaceSelected`?

Reply:
It's not hard, but I didn't want to increase the scope of the story. Also, having the whole document and the selected part on the extension side allows to replace it there.

Reply:
Anyway, the list of instructions could be extended if needed.